README.md

```diff
@@ -31,86 +31,104 @@ Umbra is a protocol for enabling stealth payments on the Ethereum blockchain. It
+
+On chain, we see Alice pays 1,000 Dai to a new empty address each week. Behind the scenes, Bob controls the keys to each of these addresses via Umbra, but nobody except Alice and Bob knows this.
 
-**What does Umbra do?**
+Bob uses Umbra to withdraw his 1,000 Dai each week. He only needs to provide an address to send it to. It’s best for him to use an address that’s not tied to his identity. He usually chooses to send it straight to an exchange, where he sells it for fiat as needed. Importantly, this means **Bob's exchange now knows this payment went to him**. To the casual chain observer— one without access to proprietary centralized exchange data— the fact that Alice's payment went to Bob is obscured.
 
-Umbra allows a payer to send funds to a fresh address. That address is controlled by the intended receiver, but only the payer and the receiver know that.
+Consider another example: Liza runs a website that asks for donations. If everyone donated by directly sending her funds, everyone would know how much Liza received in donations. If donations were sent with Umbra instead, each donation would be sent to a different address, and only Liza would know the total amount of donations she received.
 
-**How is Umbra different from Tornado Cash**
+#### How does it work?
 
-[Tornado Cash](https://tornado.cash/) is awesome and you should use it! It's an on chain mixer that uses zero knowledge proofs. You put funds in, wait a while for other people to do the same, then use your proof to withdrawal them somewhere else. Since everyone's funds are pooled in the mixer, the link between the origin address and withdrawal address is broken.
+1. When setting up your Umbra account, users sign a message. The hash of this message is used to generate two private keys—a "spending key" and a "viewing key".
+2. The corresponding public keys are both published on-chain as records associated with your ENS or CNS name.
+3. A payer uses your ENS or CNS name to look up your two public keys. Separately, the payer generates a random number.
+4. The random number is used with the spending public key to generate a "stealth address" to send funds to. The same random number is used with the viewing public key to encrypt the random number.
+5. Using the Umbra contract, the payer sends funds to the stealth address and the encrypted data is emitted as an Announcement event.
+6. The receiver scans all Announcement events from the Umbra contract. For each, they use their viewing private key to decrypt the random number, then multiply that number by their spending private key to generate the stealth private key. If the stealth private key controls the address funds were sent to, this payment was for the receiver
```

Comment:
A bit more accurate, since tokens are sent to the Umbra contract
```suggestion
6. The receiver scans all Announcement events from the Umbra contract. For each, they use their viewing private key to decrypt the random number, then multiply that number by their spending private key to generate the stealth private key. If the stealth private key controls the stealth address emitted in the Annoucement, this payment was for the receiver
```